File: Server/OnlineLibrary/OnlineLibrary.BLL/DTOs/Responses/Book/BookDetailsResponseDTO.cs
using OnlineLibrary.BLL.DTOs.Common;
using OnlineLibrary.BLL.DTOs.Responses.Author;
using OnlineLibrary.BLL.DTOs.Responses.Genre;

namespace OnlineLibrary.BLL.DTOs.Responses.Book;

public class BookDetailsResponseDTO : BookResponseDTO
{
    public GenreResponseDTO GenreResponseDTO { get; set; }
    public AuthorResponseDTO AuthorResponseDTO { get; set; }
}

File: Client/OnlineLibraryWeb/src/app/interfaces/book-details-response.dto.ts
import {GenreDTO} from "./genre.dto";
import {BookResponseDTO} from "./book-response.dto";
import {AuthorResponseDTO} from "./author.response.dto";

export interface BookDetailsResponseDTO extends BookResponseDTO {
  genreResponseDTO: GenreDTO;
  authorResponseDTO: AuthorResponseDTO;
}
